mmedit/models/editors/guided_diffusion/unet.py

```diff
@@ -0,0 +1,436 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from copy import deepcopy
+
+import torch
+import torch.nn as nn
+from mmcv.cnn.bricks.conv_module import ConvModule
+from mmengine.logging import MMLogger
+from mmengine.model import constant_init
+from mmengine.runner import load_checkpoint
+
+from mmedit.registry import MODULES
+from .modules import (DenoisingResBlock, EmbedSequential, TimeEmbedding,
+                      convert_module_to_f16, convert_module_to_f32)
+
+
+@MODULES.register_module()
+class DenoisingUnet(nn.Module):
```

Comment:
Maybe this class should inherit from `BaseModule`.

Reply:
Can checkpoints from improved-ddpm fit this class by changing config?

Reply:
> Maybe this class should inherit from `BaseModule`.

Done.

Reply:
> Can checkpoints from improved-ddpm fit this class by changing config?

Refer to https://mmengine.readthedocs.io/zh_CN/latest/advanced_tutorials/initialize.html#id3, load weights with certain prefix seems not implemented.